libelement/src/object_model/intermediaries/struct_instance.cpp

```diff
@@ -84,13 +88,18 @@ std::shared_ptr<const instruction> struct_instance::to_instruction() const
         const auto field = fields.at(input.get_name());
         assert(field);
 
-        auto expr = field->to_instruction();
-        if (!expr)
-            return nullptr;
+        auto expr = field->to_instruction(interpreter);
+        if (!expr) {
```

Comment:
if `to_instruction` returns nullptr it means that it can't be converted to an instruction. e.g if the struct instance stores a lambda in a field, since lambdas can't be represented as instructions. I don't think this is strictly an error, there are cases where we check if something can be represented as an instruction and if not, we attempt to handle it another way. [(such as for loops)](https://github.com/ultraleap/Element/blob/49e514a0135ea44726dc269f31ee81d2e97aa45d/libelement/src/object_model/intrinsics/intrinsic_for.cpp#L148)

if it's already cached then it returns the cached instruction :)